protocols/discovery/src/substream.rs

```diff
@@ -366,17 +366,12 @@ impl Substream {
             sender: context.control().clone(),
         };
         let listen_port = if context.session.ty.is_outbound() {
-            let local = multiaddr_to_socketaddr(&context.session.address)
-                .unwrap()
-                .ip()
-                .is_loopback();
-
             context
                 .listens()
                 .iter()
                 .map(|address| multiaddr_to_socketaddr(address).unwrap())
                 .filter_map(|address| {
-                    if local || RawAddr::from(address).is_reachable() {
+                    if !address.ip().is_loopback() || RawAddr::from(address).is_reachable() {
```

Comment:
https://github.com/nervosnetwork/p2p/blob/master/src/utils.rs#L19
Already checked by RawAddr::is_reachable() ?

Reply:
we need `0.0.0.0:port` and `global_ip:port`, `is_reachable` will refuse `0.0.0.0`